ledger/src/blockstore.rs

```diff
@@ -1073,7 +1149,7 @@ impl Blockstore {
         // Commit step: commit all changes to the mutable structures at once, or none at all.
         // We don't want only a subset of these changes going through.
         let path = self.coding_shred_path(slot, shred_index);
-        Self::write_all(&path, &shred.payload)?;
+        Self::rename(&path)?;
```

Comment:
hmmm so from my understanding, `write_tmp` writes the actual shreds, but that's called from `insert_shreds`, not `insert_shreds_handle_duplicate`, and window service calls the latter: https://github.com/solana-labs/solana/blob/master/core/src/window_service.rs#L127

Reply:
There is a replay attack check at the socket